src/integration/resources/inprocess/dbnode_test.go

```diff
@@ -0,0 +1,201 @@
+	)
+	req := &rpc.WriteRequest{
+		NameSpace: resources.UnaggName,
+		ID:        id,
+		Datapoint: &rpc.Datapoint{
+			Timestamp: ts.Unix(),
+			Value:     val,
+		},
+	}
+	require.NoError(t, dbnode.WritePoint(req))
+
+	freq := &rpc.FetchRequest{
+		RangeStart: ts.Add(-1 * time.Minute).Unix(),
+		RangeEnd:   ts.Add(1 * time.Minute).Unix(),
+		NameSpace:  resources.UnaggName,
+		ID:         id,
+	}
+	res, err := dbnode.Fetch(freq)
+	require.NoError(t, err)
+
+	require.Equal(t, 1, len(res.Datapoints))
+	require.Equal(t, rpc.FetchResult_{
+		Datapoints: []*rpc.Datapoint{
+			{Timestamp: ts.Unix(), Value: val},
+		},
+	}, res.Datapoints[0])
+}
+
+func TestWriteTaggedFetchTaggedRoundtrip(t *testing.T) {
+	dbnode, closer := setupNode(t)
+	defer closer()
+
+	var (
+		id  = "fooTagged"
+		ts  = time.Now()
+		val = 1.0
+	)
+	req := &rpc.WriteTaggedRequest{
+		NameSpace: resources.UnaggName,
+		ID:        id,
+		Datapoint: &rpc.Datapoint{
+			Timestamp:         ts.UnixNano(),
+			TimestampTimeType: rpc.TimeType_UNIX_NANOSECONDS,
+			Value:             val,
+		},
+		Tags: []*rpc.Tag{
+			{Name: "__name__", Value: id},
+			{Name: "job", Value: "bar"},
+		},
+	}
+	require.NoError(t, dbnode.WriteTaggedPoint(req))
+
+	query := idx.NewTermQuery([]byte("job"), []byte("bar"))
+	encoded, err := idx.Marshal(query)
+	require.NoError(t, err)
+
+	freq := &rpc.FetchTaggedRequest{
+		RangeStart:    ts.Add(-1 * time.Minute).UnixNano(),
+		RangeEnd:      ts.Add(1 * time.Minute).UnixNano(),
+		NameSpace:     []byte(resources.UnaggName),
+		RangeTimeType: rpc.TimeType_UNIX_NANOSECONDS,
+		FetchData:     true,
+		Query:         encoded,
+	}
+	res, err := dbnode.FetchTagged(freq)
+	require.NoError(t, err)
+
+	require.Equal(t, 1, len(res.Elements))
+	require.Equal(t, id, string(res.Elements[0].ID))
+	require.Equal(t, resources.UnaggName, string(res.Elements[0].NameSpace))
+
+	// Validate Tags
+	testTagDecoderPool := serialize.NewTagDecoderPool(
+		serialize.NewTagDecoderOptions(serialize.TagDecoderOptionsConfig{}),
+		pool.NewObjectPoolOptions())
+	testTagDecoderPool.Init()
+
+	dec := testTagDecoderPool.Get()
+	dec.Reset(checked.NewBytes(res.Elements[0].EncodedTags, nil))
+
+	require.True(t, dec.Next())
+	validateTag(t, dec.Current(), "__name__", id)
+	require.True(t, dec.Next())
+	validateTag(t, dec.Current(), "job", "bar")
+	require.False(t, dec.Next())
+}
+
+func validateTag(t *testing.T, tag ident.Tag, name string, value string) {
+	require.Equal(t, name, tag.Name.String())
+	require.Equal(t, value, tag.Value.String())
+}
+
+func setupNode(t *testing.T) (resources.Node, func()) {
+	dbnode, err := NewDBNodeFromYAML(defaultDBNodeConfig, DBNodeOptions{})
+	require.NoError(t, err)
+
+	coord, err := NewCoordinatorFromYAML(defaultCoordConfig, CoordinatorOptions{})
+	require.NoError(t, err)
+
+	require.NoError(t, coord.WaitForNamespace(""))
+
+	host, err := dbnode.HostDetails(9000)
+	require.NoError(t, err)
+
+	_, err = coord.CreateDatabase(admin.DatabaseCreateRequest{
```

Comment:
Would also be nice to simplify database creation as well after we get the foundational stuff in place.